packages/gatsby/src/utils/cache.ts

```diff
@@ -0,0 +1,77 @@
+import manager, { Store, StoreConfig, CachingConfig } from "cache-manager"
+import fs from "fs-extra"
+import fsStore from "cache-manager-fs-hash"
+import path from "path"
+
+const MAX_CACHE_SIZE = 250
+const TTL = Number.MAX_SAFE_INTEGER
+
+interface ICacheProperties {
+  name?: string
+  store?: Store
+}
+
+class Cache {
+  public name: string
+  public store: Store
+  public cache?: manager.Cache
+
+  constructor({ name = `db`, store = fsStore }: ICacheProperties = {}) {
+    this.name = name
+    this.store = store
+  }
+
+  get directory(): string {
+    return path.join(process.cwd(), `.cache/caches/${this.name}`)
+  }
+
+  init(): Cache {
+    fs.ensureDirSync(this.directory)
+
+    const configs: StoreConfig[] = [
+      {
+        store: `memory`,
+        max: MAX_CACHE_SIZE,
+        ttl: TTL,
+      },
+      {
+        store: this.store,
+        ttl: TTL,
```

Comment:
cache-manager says this is a required proeprty, even though the fs-hash store uses the one inside options